external-crates/move/crates/move-regex-borrow-graph/src/references.rs

```diff
@@ -0,0 +1,436 @@
+// Copyright (c) The Move Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    bail,
+    collections::{Path, Paths},
+    error,
+    regex::Regex,
+    Result,
+};
+use itertools::Either;
+use std::{
+    cmp,
+    collections::{BTreeMap, BTreeSet},
+    fmt::{self, Debug},
+};
+
+//**************************************************************************************************
+// Definitions
+//**************************************************************************************************
+
+#[derive(Clone, Copy, Debug, Eq, PartialEq, Ord, PartialOrd)]
+pub struct Ref(Ref_);
```

Comment:
Why is `Ref` newtyped?

Reply:
To not expose the variants outside of this module